### DISCUSSION TITLE
Support for common output formats?

### DISCUSSION THREAD

**Discussion by BHSPitMonkey** [github-linguist/linguist#6363] [General]
Is this project open to supporting highlighting for common _output_ formats (from tools like `strace`, `dmesg`, `journalctl` etc.) which (1) aren't true "languages" and (2) don't usually appear in repositories as standalone files?

I was sharing some `strace` output in a GitHub Issue today via a Markdown code block and had hoped that using a fence like ` ```strace ` would add some nice visual highlighting to the output, but I did not find a way to do this. vim, for example, supports strace output highlighting (defined [here](https://github.com/vim/vim/blob/master/runtime/syntax/strace.vim)). In case you have no idea what this output looks like, here's a sample:

```
pselect6(0, NULL, NULL, NULL, {tv_sec=1, tv_nsec=0}, NULL) = 0 (Timeout)
ioctl(3, TCFLSH, TCIFLUSH)              = 0
write(3, "\32\3008\274~", 5)            = 5
pselect6(5, [3 4], [], [], NULL, NULL)  = 1 (in [3])
read(3, "\32\301\2\v\nR~", 7)           = 7
write(3, "\0B!\250\\,\240~", 8)         = 8
pselect6(7, [6], [3], [], NULL, NULL)   = 1 (out [3])
```

The policy stated in this project's [CONTRIBUTING.md](https://github.com/github/linguist/blob/master/CONTRIBUTING.md) is based on the frequency in which files in GitHub repos exist, but cases like this would not seem to meet that criteria despite being very useful when presenting data within Markdown blobs in Issues, Wiki pages, comments, etc. A similar use case I use often is ` ```diff `, but diffs have the benefit of appearing often as standalone files (usually with the `.patch` extension).

I think opening up to these use cases would be a great readability/accessibility win in lots of places where Markdown is used across GitHub. What do others think?

**Comment by lildude** [comment]
> Is this project open to supporting highlighting for common output formats (from tools like `strace`, `dmesg`, `journalctl` etc.) which (1) aren't true "languages" and (2) don't usually appear in repositories as standalone files?

We certainly are, and have done so very recently with Elvish Transcript added in https://github.com/github/linguist/pull/6302 - that's the easy bit 😁 The hardest bit is finding a TextMate/VSCode-compatible grammar for the output formats as, as you've stated, they're not really languages traditionally written in a code editor, so don't often have TextMate/VSCode-compatible grammars written for them.

If you can find such a grammar, and can prove similar usage requirements within markdown files and issue and PR comments, we'd certainly consider a PR.

One thing to keep in mind is Linguist is only responsible for providing the grammars and associating them with a "language" used by these blocks. The actual rendering is handled by github/markup. See my comment at https://github.com/github/linguist/pull/6302#issuecomment-1453261600
